src/js/yeldaWebchatInjector.js

```diff
@@ -786,11 +793,43 @@ class YeldaChat {
     return true
   }
 
+  /**
+   * Check if the data currently handled is outdated
+   * 
+   * If this.assistantId or this.locale (new data is different from the data.assistantId or data.locale (previous data)
+   * it means that the user has changed the selected locale or the selected assistant quickly 
+   * before the chat finished loading with the previous data
+   * So we can stop the current process to let the chat load with the new data
+   * 
+   * @param {Object} data - iframe data sent from webchat window
+   * @param {String} data.settingId
+   * @param {String} data.assistantId
+   * @param {String} data.locale
+   * @param {Object} globalData
+   * @param {String} globalData.settingId
+   * @param {String} globalData.assistantId
+   * @param {String} globalData.locale
+   * @returns {Boolean}
+   */
+  isDataOutdated(data, { settingId, assistantId, locale }) {
+    if (data.settingId && data.settingId !== settingId) {
+      return true
+    }
+
+    if (!data.settingId && (data.assistantId !== assistantId || data.locale !== locale)) {
+      return true
+    }
+    return false
+  }
```

Comment:
i took the liberty to make suggestions 
- "   * it means that the user has changed the selected locale or the selected assistant quickly "
The use case does not really interest us here (switching locale too fast), the real point the is that we want to handle a race condition, whatever the trigger. " multiple new webchat loading requests are done in a short amount of time, " is enough for example
- "So we can stop the current process to let the chat load with the new data" => new data is not very clear in our context
- "   * @param {String} data.settingId" => data is quite wild, i suggest we used a more precised parameter name ?

```suggestion
  /**
   * Return if the webchat loadChat settings are different from the yeldaChat global assistant settings
   * 
   * -  yeldaChat can be init (and reset) either with settingId parameter or the couple (assistantId, locale) that we store globally 
   * - Each new webchat loading request implies a "getAssistantSettings" async API call, 
   * => If multiple new webchat loading requests are done in a short amount of time, 
   * "getAssistantSettings" response can be subject to race condition issue : 
   * The latest requested answer received might not be the latest API call answer.
   * 
   * isDataOutdated return if one of the global parameter is different from the webchat settings we got from "getAssistantSettings" API call
   * so "loadChat" function can be informed that the latest request is not the one currently handled
   * And can stop the current process to let the chat load with the global last requested settings
   * 
   * @param {Object} webchatSettings - current settings from the reset or init yeldaChat request
   * @param {String} webchatSettings.settingId
   * @param {String} webchatSettings.assistantId
   * @param {String} webchatSettings.locale
   * @param {Object} globalData - latest globally stored settings from the reset or init yeldaChat request
   * @param {String} globalData.settingId
   * @param {String} globalData.assistantId
   * @param {String} globalData.locale
   * @returns {Boolean}
   */
  isDataOutdated(data, { settingId, assistantId, locale }) {
    if (data.settingId && data.settingId !== settingId) {
      return true
    }

    if (!data.settingId && (data.assistantId !== assistantId || data.locale !== locale)) {
      return true
    }
    return false
  }
```